packages/fetchai/skills/coin_price/handlers.py

```diff
@@ -0,0 +1,192 @@
+            self.context.logger.info(
+                "got unexpected http message: code = " + str(message.status_code)
+            )
+
+    def _handle_request(
+        self, http_msg: HttpMessage, http_dialogue: HttpDialogue
+    ) -> None:
+        """
+        Handle a Http request.
+
+        :param http_msg: the http message
+        :param http_dialogue: the http dialogue
+        :return: None
+        """
+        self.context.logger.info(
+            "received http request with method={}, url={} and body={!r}".format(
+                http_msg.method, http_msg.url, http_msg.body,
+            )
+        )
+        if http_msg.method == "get":
+            self._handle_get(http_msg, http_dialogue)
+        elif http_msg.method == "post":
+            self._handle_post(http_msg, http_dialogue)
+
+    def _handle_get(self, http_msg: HttpMessage, http_dialogue: HttpDialogue) -> None:
+        """
+        Handle a Http request of verb GET.
+
+        :param http_msg: the http message
+        :param http_dialogue: the http dialogue
+        :return: None
+        """
+        http_response = http_dialogue.reply(
+            performative=HttpMessage.Performative.RESPONSE,
+            target_message=http_msg,
+            version=http_msg.version,
+            status_code=200,
+            status_text="Success",
+            headers=http_msg.headers,
+            body=json.dumps(self.context.shared_state["oracle_data"]).encode("utf-8"),
+        )
+        self.context.logger.info("responding with: {}".format(http_response))
+        envelope_context = EnvelopeContext(connection_id=HTTP_SERVER_ID)
+        self.context.outbox.put_message(message=http_response, context=envelope_context)
+
+    def _handle_post(self, http_msg: HttpMessage, http_dialogue: HttpDialogue) -> None:
+        """
+        Handle a Http request of verb POST.
+
+        :param http_msg: the http message
+        :param http_dialogue: the http dialogue
+        :return: None
+        """
+        http_response = http_dialogue.reply(
+            performative=HttpMessage.Performative.RESPONSE,
+            target_message=http_msg,
+            version=http_msg.version,
+            status_code=200,
+            status_text="Success",
+            headers=http_msg.headers,
+            body=b"",
+        )
+        self.context.logger.info("responding with: {}".format(http_response))
+        self.context.outbox.put_message(message=http_response)
+
+    def _handle_unidentified_dialogue(self, msg: Message) -> None:
+        """
+        Handle an unidentified dialogue.
+
+        :param msg: the unidentified message to be handled
+        :return: None
+        """
+
+        self.context.logger.info(
+            "received invalid message={}, unidentified dialogue.".format(msg)
+        )
+
+    def teardown(self) -> None:
+        """
+        Teardown the handler.
+
+        :return: None
+        """
+        self.context.logger.info("tearing down HttpHandler")
```

Comment:
We avoid these now on the info level. It makes it too noisy. On the debug level there is already a log for each teardown call. So `pass` is probably best here.